### PR TITLE
feat: Updated README.md, defaultCodeModularity, added submodules

### PR DESCRIPTION
**Pull request by ZaBarudo**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "codeMapConfig"]
```
Comment (Ram-20062003) [inline .gitmodules]:
```suggestion
[submodule "player_code"]
```

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+	path = server/src/main/resources/codeMapConfig
```
Comment (Ram-20062003) [inline .gitmodules]:
```suggestion
	path = server/src/main/resources/player_code
```

File: README.md [hunk 3]
```diff
@@ -24,7 +24,9 @@
 3. Press `Ctrl+Alt+Shift+S` and make sure the JDK version is 17.
 4. Press `Ctrl+Alt+S` and go to `Build, Execution, Deployment -> Docker` and make sure docker is configured correctly/
 5. Copy `server/src/main/resources/application.example.yml` to `server/src/main/resources/application.yml`. If you want to use docker instead, copy the `server/src/main/resources/application.docker.example.yml` to `server/src/main/resources/application.yml`.
-6. The run configurations will be available in the top bar:
+6. Run `git submodule update --init` or `git submodule add --name codeMapConfig https://github.com/delta/codecharacter-default-codes-2022 server/src/main/resources/codeMapConfig`
```
Comment (Ram-20062003) [inline README.md]:
```suggestion
6. Run `git submodule update --init` 
```

File: server/src/main/kotlin/delta/codecharacter/server/config/DefaultCodeMapConfiguration.kt [hunk 4]
```diff
@@ -9,106 +9,7 @@ import java.time.Instant
...
-            "  for (auto &[type_id, pos] : game.get_spawn_positions()) {\n" +
-            "    // you can use logger macro as well, which is an alias for game.logr()\n" +
-            "    logger << \"Type \" << type_id << \" at Position (\" << pos.get_x() << \",\"\n" +
-            "           << pos.get_y() << \")\\n\";\n" +
-            "  }\n" +
-            "\n" +
-            "  // always  return the game object\n" +
-            "  return game;\n" +
-            "}\n",
+        {}::class.java.classLoader.getResource("codeMapConfig/cpp/run.cpp")?.readText() ?: "",
```
Comment (Ram-20062003) [inline server/src/main/kotlin/delta/codecharacter/server/config/DefaultCodeMapConfiguration.kt]:
```suggestion
        DefaultCodeMapConfiguration::class.java.classLoader.getResource("codeMapConfig/cpp/run.cpp")?.readText() ?: "",
```

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]

**Comment by Ram-20062003** [comment]
While updating .gitmodules, please clear the previous .git/modules and in .git/config

File: .gitmodules [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[submodule "player_code"]
```
Comment (Ram-20062003) [inline .gitmodules:1]: - While updating .gitmodules, please clear the previous .git/modules and in .git/config

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Review by Ashwin125 — APPROVED** [review]

**Pull request merged**